client.go

```diff
@@ -205,151 +201,99 @@ func (r *marathonClient) Ping() (bool, error) {
+	return r.apiCall("DELETE", uri, post, result)
 }
 
-func (r *marathonClient) apiOperation(method, uri string, post, result interface{}) error {
-	content, err := r.buildPostData(post)
+func (r *marathonClient) apiCall(method, uri string, body, result interface{}) error {
+	// Get a member from the cluster
+	marathon, err := r.cluster.GetMember()
 	if err != nil {
 		return err
 	}
 
-	_, _, err = r.apiCall(method, uri, content, result)
-
-	return err
-}
-
-func (r *marathonClient) apiCall(method, uri, body string, result interface{}) (int, string, error) {
-	glog.V(debugLevel).Infof("[api]: method: %s, uri: %s, body: %s", method, uri, body)
-
-	status, content, _, err := r.httpRequest(method, uri, body)
-	if err != nil {
-		return 0, "", err
-	}
+	url := fmt.Sprintf("%s/%s", marathon, uri)
 
-	glog.V(debugLevel).Infof("[api] result: status: %d, content: %s\n", status, content)
-	if status >= 200 && status <= 299 {
-		if result != nil {
-			if err := r.decodeRequest(strings.NewReader(content), result); err != nil {
-				glog.V(debugLevel).Infof("failed to unmarshall the response from marathon, error: %s", err)
-				return status, content, ErrInvalidResponse
-			}
+	var jsonBody []byte
+	if body != nil {
+		jsonBody, err = json.Marshal(body)
+		if err != nil {
+			return err
 		}
-		return status, content, nil
 	}
 
-	switch status {
-	case 500:
-		return status, "", ErrInvalidResponse
-	case 404:
-		return status, "", ErrDoesNotExist
-	case 409:
-		return status, "", ErrConflict
+	// Make the http request to Marathon
+	request, err := http.NewRequest(method, url, bytes.NewReader(jsonBody))
+	if err != nil {
+		return err
 	}
 
-	// step: lets decode into a error message
-	var message struct {
-		Message string `json:"message"`
+	// Add any basic auth and the content headers
+	if r.config.HTTPBasicAuthUser != "" {
+		request.SetBasicAuth(r.config.HTTPBasicAuthUser, r.config.HTTPBasicPassword)
 	}
+	request.Header.Add("Content-Type", "application/json")
+	request.Header.Add("Accept", "application/json")
 
-	if err := r.decodeRequest(strings.NewReader(content), &message); err != nil {
-		return status, content, ErrInvalidResponse
+	response, err := r.httpClient.Do(request)
+	if err != nil {
+		return err
 	}
+	defer response.Body.Close()
 
-	errorMessage := "unknown error"
-	if message.Message != "" {
-		errorMessage = message.Message
+	respBody, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return err
 	}
 
-	return status, "", fmt.Errorf("%s", errorMessage)
-}
-
-func (r *marathonClient) httpRequest(method, uri, body string) (int, string, *http.Response, error) {
-	var content string
-	var response *http.Response
+	if len(jsonBody) > 0 {
+		log.Printf("apiCall(): %v %v %s returned %v %s\n", request.Method, request.URL.String(), jsonBody, response.Status, oneLogLine(respBody))
+	} else {
+		log.Printf("apiCall(): %v %v returned %v %s\n", request.Method, request.URL.String(), response.Status, oneLogLine(respBody))
```

Comment:
Another question: how to make it silent now? Before it was possible with setting debug level, now it will log every time.
IMO in this case we should either remove debugging statements or put the debugging mode into the config.